components/dashboard/src/org-admin/RunningWorkspacesCard.tsx

```diff
@@ -0,0 +1,199 @@
+            from: Timestamp.fromDate(dayjs().subtract(lookbackHours, "hours").startOf("day").toDate()),
+        });
+
+    useEffect(() => {
+        if (hasNextPage && !isFetchingNextPage) {
+            fetchNextPage();
+        }
+    }, [hasNextPage, isFetchingNextPage, fetchNextPage]);
+
+    const runningWorkspaces = useMemo(() => {
+        if (!data?.pages) {
+            return [];
+        }
+        const allSessions = data.pages.flatMap((page) => page);
+        return allSessions.filter(isWorkspaceNotStopped);
+    }, [data]);
+
+    const handleStopAllWorkspaces = async () => {
+        if (runningWorkspaces.length === 0) {
+            toast.toast({ type: "error", message: "No running workspaces to stop." });
+            setIsStopAllModalOpen(false);
+            return;
+        }
+
+        setIsStoppingAll(true);
+        let successCount = 0;
+        let errorCount = 0;
+
+        const stopPromises = runningWorkspaces.map(async (session) => {
+            if (session.workspace?.id) {
+                try {
+                    await workspaceClient.stopWorkspace({ workspaceId: session.workspace.id });
+                    successCount++;
+                } catch (e) {
+                    console.error(`Failed to stop workspace ${session.workspace.id}:`, e);
+                    errorCount++;
+                }
+            }
+        });
+
+        await Promise.allSettled(stopPromises);
+
+        setIsStoppingAll(false);
+        setIsStopAllModalOpen(false);
+
+        if (errorCount > 0) {
+            toast.toast({
+                type: "error",
+                message: `Failed to stop all workspaces`,
+                description: `Attempted to stop ${runningWorkspaces.length} workspaces. ${successCount} stopped, ${errorCount} failed.`,
+            });
+        } else {
+            toast.toast({
+                type: "success",
+                message: `Stop command sent`,
+                description: `Successfully sent stop command for ${successCount} workspaces.`,
+            });
+        }
+        refetch();
+    };
+
+    if (isLoading && !isStoppingAll) {
+        return (
+            <div className="flex items-center justify-center w-full space-x-2 text-gray-400 text-sm p-8">
+                <img alt="Loading Spinner" className="h-4 w-4 animate-spin" src={Spinner} />
+                <span>Loading running workspaces...</span>
+            </div>
+        );
+    }
+
+    if (isError && error) {
+        return (
+            <Alert type="error" className="m-4">
+                <p>Error loading running workspaces:</p>
+                <pre>{error instanceof Error ? error.message : String(error)}</pre>
+            </Alert>
+        );
+    }
+
+    return (
+        <div className="bg-white dark:bg-gray-800 shadow-md rounded-lg p-4 mt-6">
+            <div className="flex justify-between items-center mb-3">
+                <h3 className="text-lg font-semibold text-gray-700 dark:text-gray-200">
```

Comment:
use `Heading3` from podkit for MAXIMUM CONSISTENCY 💥

Reply:
Done